kotter_report.py

```diff
@@ -1,187 +1,240 @@
 #!/usr/bin/env /home/epetz/.cache/pypoetry/virtualenvs/weaselbot-7wWSi8jP-py3.11/bin/python3.11
 
-import os
-import ssl
+import logging
 from datetime import date, datetime, timedelta
+from typing import Any
 
 import numpy as np
 import pandas as pd
-from dotenv import load_dotenv
 from slack_sdk import WebClient
-from sqlalchemy import create_engine
-
-os.chdir(os.path.dirname(os.path.abspath(__file__)))
-
-# Will need to use PAXMiner creds
-dummy = load_dotenv()
-DATABASE_USER = os.environ.get("DATABASE_USER")
-DATABASE_PASSWORD = os.environ.get("DATABASE_PASSWORD")
-DATABASE_HOST = os.environ.get("DATABASE_HOST")
-engine = create_engine(f"mysql+mysqlconnector://{DATABASE_USER}:{DATABASE_PASSWORD}@{DATABASE_HOST}:3306")
-
-# Inputs
-year_select = date.today().year
-no_post_threshold = 2
-reminder_weeks = 2
-home_ao_capture = datetime.combine(
-    date.today() + timedelta(weeks=-8), datetime.min.time()
-)  # pulls the last 8 weeks to determine home AO
-no_q_threshold_weeks = 4
-no_q_threshold_posts = 4
-active_post_threshold = 3
-# db = 'f3stcharles'
-# paxminer_log_channel = 'C123'
+from slack_sdk.errors import SlackApiError
+from sqlalchemy import MetaData, Table
+from sqlalchemy.engine import Engine
+from sqlalchemy.sql import Selectable, and_, case, func, or_, select
+
+from utils import mysql_connection, slack_client
+
+NO_POST_THRESHOLD = 2
+REMINDER_WEEKS = 2
+HOME_AO_CAPTURE = datetime.combine(date.today() + timedelta(weeks=-8), datetime.min.time())
+NO_Q_THRESHOLD_WEEKS = 4
+NO_Q_THRESHOLD_POSTS = 4
```

Comment:
Just for awareness, I've had requests to make this region-configurable... really wouldn't be hard at all, just load add these fields to the region config table.

Reply:
I saw those comments too. For now, it could be set up as a test run for those regions specifically asking for it. The logic to test whether or not the config table is even there is trivial. Default would be the values above if not table. Otherwise, override with the table values. Let's talk through how a region config table should look and we can try it out.